pdns/dnsdist.cc

```diff
@@ -1801,14 +1797,18 @@ static void checkFileDescriptorsLimits(size_t udpBindsCount, size_t tcpBindsCoun
   }
   requiredFDsCount += backendUDPSocketsCount;
   /* TCP sockets to backends */
-  requiredFDsCount += (backends->size() * g_maxTCPClientThreads);
+  if (g_maxTCPClientThreads) {
+    requiredFDsCount += (backends->size() * (*g_maxTCPClientThreads));
+  }
   /* listening sockets */
   requiredFDsCount += udpBindsCount;
   requiredFDsCount += tcpBindsCount;
-  /* max TCP connections currently served */
-  requiredFDsCount += g_maxTCPClientThreads;
-  /* max pipes for communicating between TCP acceptors and client threads */
-  requiredFDsCount += (g_maxTCPClientThreads * 2);
+  /* number of TCP connections currently served, assuming 1 connection per worker thread which is of course not right */
+  if (g_maxTCPClientThreads) {
+    requiredFDsCount += *g_maxTCPClientThreads;
```

Comment:
This looks weird... why adding three times `*g_maxTCPClientThreads` ?

Reply:
It should always be set at this point, I just check it to prevent a crash if I'm wrong :)